blaze/compute/python.py

```diff
@@ -666,6 +667,12 @@ def compute_up(t, seq, **kwargs):
         return take(t.n, seq)
 
 
+@dispatch(Sample, Sequence)
+def compute_up(t, seq, **kwargs):
+    nsamp = t.n if t.n is not None else int(t.frac * len(seq))
+    return random.sample(seq, min(nsamp, len(seq)))
```

Comment:
I wonder if an optimization might be to just be identity if `n >= len(seq)`. is the contract that `sample` is in random order or just that it is a random sampling?